File: app/src/main/java/com/kota/Telnet/Logic/SearchBoard_Handler.java
package com.kota.Telnet.Logic;

import com.kota.Telnet.TelnetClient;
import java.util.Vector;

public class SearchBoard_Handler {
    private static SearchBoard_Handler _instance = null;
    private Vector<String> _boards = new Vector<>();

    private SearchBoard_Handler() {
    }

    public static SearchBoard_Handler getInstance() {
        if (_instance == null) {
            _instance = new SearchBoard_Handler();
        }
        return _instance;
    }

    public void read() {
        int i = 3;
        while (i < 23) {
            String content = TelnetClient.getModel().getRowString(i).trim();
            if (content.length() != 0) {
                for (String board : content.split(" +")) {
                    if (board.length() > 0) {
                        this._boards.add(board);
                    }
                }
                i++;
            } else {
                return;
            }
        }
    }

    public void clear() {
        this._boards.clear();
    }

    public int getBoardsSize() {
        return this._boards.size();
    }

    public String getBoard(int index) {
        return this._boards.get(index);
    }

    public String[] getBoards() {
        return (String[]) this._boards.toArray(new String[this._boards.size()]);
    }
}

File: app/src/main/java/com/kota/Telnet/Logic/SearchBoard_Handler.kt
package com.kota.Telnet.Logic

import com.kota.Telnet.TelnetClient
import java.util.Vector

class SearchBoard_Handler private constructor() {
    private val _boards = Vector<String?>()

    fun read() {
        var i = 3
        while (i < 23) {
            val content = TelnetClient.getModel().getRowString(i).trim { it <= ' ' }
            if (content.length != 0) {
                for (board in content.split(" +".toRegex()).dropLastWhile { it.isEmpty() }
                    .toTypedArray()) {
                    if (board.length > 0) {
                        this._boards.add(board)
                    }
                }
                i++
            } else {
                return
            }
        }
    }

    fun clear() {
        this._boards.clear()
    }

    val boardsSize: Int
        get() = this._boards.size

    fun getBoard(index: Int): String? {
        return this._boards.get(index)
    }

    val boards: Array<String?>
        get() = this._boards.toTypedArray<String?>()

    companion object {
        private var _instance: SearchBoard_Handler? = null
        @JvmStatic
        val instance: SearchBoard_Handler
            get() {
                if (_instance == null) {
                    _instance = SearchBoard_Handler()
                }
                return _instance!!
            }
    }
}
